src/views/transaction/transaction.view.jsx

```diff
@@ -139,104 +142,74 @@ function Transaction ({
-                                  {transactionTask.data.fromAccountIndex}
+                                <Link to={`/user-account/${transactionTask.data.fromBjj ? transactionTask.data.fromBjj : transactionTask.data.fromBJJ}`}>
+                                  {transactionTask.data.fromBjj ? transactionTask.data.fromBjj : transactionTask.data.fromBJJ}
                                 </Link>
                               </Col>
                             </Row>
                           </Col>
                         </Row>
                         )
                       : <></>}
-                    {!transactionTask.data.fromHezEthereumAddress && !transactionTask.data.fromBjj && transactionTask.data.fromAccountIndex
-                      ? (
-                        <Row>
-                          <Col>From</Col>
-                          <Col>
-                            <Row wrapped>
-                              <CopyToClipboardButton content={transactionTask.data.fromAccountIndex} />
-                              <Col wrapped>
-                                {transactionTask.data.fromAccountIndex}
-                              </Col>
-                            </Row>
-                          </Col>
-                        </Row>
-                        )
-                      : <></>}
-                    {transactionTask.data.toHezEthereumAddress
+                    {transactionTask.data.toHezEthereumAddress && transactionTask.data.toHezEthereumAddress !== INTERNAL_ACCOUNT_ETH_ADDR
                       ? (
                         <Row>
                           <Col>To</Col>
                           <Col link>
                             <Row wrapped>
-                              <CopyToClipboardButton content={transactionTask.data.type === 'Exit' ? transactionTask.data.toAccountIndex : transactionTask.data.toHezEthereumAddress} />
+                              <CopyToClipboardButton content={transactionTask.data.toHezEthereumAddress} />
                               <Col wrapped>
                                 <Link to={`/user-account/${transactionTask.data.toHezEthereumAddress}`}>
-                                  {transactionTask.data.type === 'Exit' ? transactionTask.data.toAccountIndex : transactionTask.data.toHezEthereumAddress}
+                                  {transactionTask.data.toHezEthereumAddress}
                                 </Link>
                               </Col>
                             </Row>
                           </Col>
                         </Row>
                         )
                       : <></>}
-                    {!transactionTask.data.toHezEthereumAddress && transactionTask.data.toBjj
+                    {(!transactionTask.data.toHezEthereumAddress || transactionTask.data.toHezEthereumAddress === INTERNAL_ACCOUNT_ETH_ADDR) && (transactionTask.data.toBjj || transactionTask.data.toBJJ)
```

Comment:
I think this line which is also repeated above can be extracted to a function as it's the most confusing one